src/main/java/edu/kit/provideq/toolbox/SolveRequest.java

```diff
@@ -1,13 +1,30 @@
 package edu.kit.provideq.toolbox;
 
+import edu.kit.provideq.toolbox.meta.ProblemType;
 import org.springframework.lang.Nullable;
 
+import java.util.Map;
+
 /**
- * Abstract solve request containing some request content
+ * Solve request containing some request content
+ * CANNOT be abstract, otherwise internal springboot initializer throws
```

Comment:
The problem is that this is the base type that is used in all REST controllers.
For example in SAT:
`public SolutionHandle solveSat(@RequestBody @Valid SolveSatRequest request)`

The incoming data is automatically parsed to the type, and it threw an exception for me in that case.

It can't be a record, because Java doesn't allow record inheritance :(